flink-python/pyflink/datastream/state.py

```diff
@@ -316,6 +316,10 @@ def __init__(self, name: str, value_type_info: TypeInformation):
         :param name: The name of the state.
         :param value_type_info: the type information of the state.
         """
+        if not isinstance(value_type_info, PickledBytesTypeInfo):
```

Comment:
Move this logic to the `StateDescriptor`? After that,  we can remove the duplicate code in every child StateDescriptor